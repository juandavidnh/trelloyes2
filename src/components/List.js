import React from 'react';
import './List.css';
import Card from './Card';

function List(props){
    const cardsArray = props.cards;


    return (
        <section className="List">
            <header className="List-header">
                <h2>{props.header}</h2>
            </header>
            <div className="List-cards">
                {cardsArray.map(
                    (card)=>(
                        <Card 
                            key={card.id}
                            title={card.title}
                            content={card.content}
                        />
                    )
                )}
                <button type="button" className="List-add-button">
                    + Add Random Card
                </button>
            </div>
        </section>
    )
}

export default List;