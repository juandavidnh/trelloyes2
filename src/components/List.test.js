import React from 'react';
import ReactDOM from 'react-dom';
import List from './List';
import renderer from 'react-test-renderer';

it('renders without crashing', () => {
  const div = document.createElement('div');
  ReactDOM.render(<List cards={[{id:10, title:'your mom', content:"don't you cry tonight"}, {id:9, title:'your mom', content:"don't you cry tonight"}]}/>, div);
  ReactDOM.unmountComponentAtNode(div);
});

it('renders the UI as expected', () => {
    const tree = renderer
      .create(<List cards={[{id:10, title:'your mom', content:"don't you cry tonight"},{id:9, title:'your mom', content:"don't you cry tonight"}]} header="this list" />)
      .toJSON();
    expect(tree).toMatchSnapshot();  
});
