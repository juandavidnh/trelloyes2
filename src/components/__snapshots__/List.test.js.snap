// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders the UI as expected 1`] = `
<section
  className="List"
>
  <header
    className="List-header"
  >
    <h2>
      this list
    </h2>
  </header>
  <div
    className="List-cards"
  >
    <div
      className="Card"
    >
      <button
        type="button"
      >
        delete
      </button>
      <h3>
        your mom
      </h3>
      <p>
        don't you cry tonight
      </p>
    </div>
    <div
      className="Card"
    >
      <button
        type="button"
      >
        delete
      </button>
      <h3>
        your mom
      </h3>
      <p>
        don't you cry tonight
      </p>
    </div>
    <button
      className="List-add-button"
      type="button"
    >
      + Add Random Card
    </button>
  </div>
</section>
`;
